.goreleaser.yml

```diff
@@ -11,6 +11,9 @@ builds:
     goarch:
     - amd64
     - arm64
+    hooks:
+      post:
+        - bash -c 'if [ "{{ .Env.GGOOS  }}" != "darwin" ]; then upx -q "{{ .Path }}" || true; fi'
```

Comment:
@dustin-decker I like the UPX optimization. Couple of comments:

1. Are we sure this works for windows? The UPX readme suggests that windows arm64 isn't supported. 
2. I find this oneliner to be a little hard on the eyes. WDYT about specifically defining a darwin build like:

```yml
builds:
  - id: trufflehog-upx
    binary: trufflehog
    ldflags:
      - -X 'github.com/trufflesecurity/trufflehog/v3/pkg/version.BuildVersion={{ .Version }}'
    env: [CGO_ENABLED=0]
    goos:
    - linux
    goarch:
    - amd64
    - arm64
    hooks:
      post:
        - upx "{{ .Path }}"
  - id: trufflehog
    binary: trufflehog
    ldflags:
      - -X 'github.com/trufflesecurity/trufflehog/v3/pkg/version.BuildVersion={{ .Version }}'
    env: [CGO_ENABLED=0]
    goos:
    - darwin
    - windows
    goarch:
    - amd64
    - arm64
 ```
This could be split out differently too like on OS rather than upx enabled - `trufflehog-linux`, `trufflehog-darwin`, `trufflehog-windows`.